RELEASE/scripts/autoscend/combat/auto_combat_default_stage2.ash

```diff
@@ -214,18 +214,70 @@ string auto_combatDefaultStage2(int round, monster enemy, string text)
 	}
 	
 	# Instakill handler
-	boolean doInstaKill = true;
-	if($monsters[Lobsterfrogman, Ninja Snowman Assassin] contains enemy)
+	boolean couldInstaKill = true;
+	if($monsters[Smut Orc Pipelayer,Smut Orc Jacker,Smut Orc Screwer,Smut Orc Nailer] contains enemy && get_property("chasmBridgeProgress").to_int() < 30)
+	{
+		//want to do cold damage in stage3
+		if(my_adventures() > 6)
+		{
+			couldInstaKill = false;
+		}
+	}
+	else if($monsters[Lobsterfrogman, Ninja Snowman Assassin] contains enemy)
 	{
 		if(auto_have_skill($skill[Digitize]) && (get_property("_sourceTerminalDigitizeMonster") != enemy))
 		{
-			doInstaKill = false;
+			couldInstaKill = false;
 		}
 	}
 
-	if(instakillable(enemy) && !isFreeMonster(enemy) && doInstaKill)
+	if(instakillable(enemy) && !isFreeMonster(enemy) && couldInstaKill)
 	{
-		if(canUse($skill[lightning strike]))
+		boolean wantInstaKill;
```

Comment:
set to false to make it clear what initial state is. Currently it is possible to use this variable without having it get set

Reply:
Same for waitForDesert and waitForCyrpt

Reply:
boolean is false if not set, is there a compiler that wants the initial value to be explicit? explicit = false is probably not harmful though